submissions/andrewklmn/friends-app/app.js

```diff
@@ -0,0 +1,358 @@
+const filterGender = document.querySelector('.filter-gender');
+const filterMinAge = document.querySelector('.filter-min-age');
+const filterMaxAge = document.querySelector('.filter-max-age');
+const filterCountry = document.querySelector('.filter-country');
+
+const drawPerson = ({location, phone, email, gender, name, picture, dob}) => {  
+  const {street, city, state, country} = location;
+  const div = document.createElement('div');
+  div.classList.add('person');
+  div.title = `Address: ${street.number}, ${street.name}, ${city}, ${state}
+  Phone: ${phone}
+  Email: ${email}`;
+  div.innerHTML =`
+      <div class="person-name ${gender}">${name.first} ${name.last}</div>
+      <div class="person-image">
+      <picture>
+        <source media="(max-width:300px)" 
+                srcset="${picture.thumbnail} 1x, ${picture.medium} 2x">
+        <source media="(max-width:400px)" 
+                srcset="${picture.medium} 1x, ${picture.large} 2x">
+        <img class="rounded" src="${picture.large}">
+      </picture>
+      </div>
+      <div class="person-age">
+        ${dob.age} y.o.
+      </div>
+      <div class="person-location">
+        ${country}
+      </div>
+  `;
+  container.appendChild(div);
+}
+
+const filterList = ({friends, filters})=>{
+  return friends.filter(({name, gender, dob, location}) => {
+    if (filters.namePart != ''
+          && !`${name.first} ${name.last}`.toUpperCase()
+                                          .includes(filters.partOfName.toUpperCase())) { 
+        return false;
+    }
+    if (filters.gender != '' 
+          && gender != filters.gender) { 
+      return false;
+    }
+    if (filters.ageRange[0] > dob.age
+        || filters.ageRange[1] < dob.age) {
+      return false;
+    }
+    if (filters.country != '') {
+      if(location.country != filters.country) { 
+        return false;
+      }
+    };
+
+    return true;
+  })
+}
+
+const sortList = (friends) => {
+
+  if(state.sorter.keyName == '') {
+    return friends;
+  }
+
+  const [index0, index1] = state.sorter.fieldPathList[state.sorter.keyName];
+  
+  return friends.sort(function(a, b){
+    if(state.sorter.keyName === '') return 0;
+    if(state.sorter.order === 'DESC') {
+      if (a[index0][index1] > b[index0][index1]) {
+        return -1;
+      }
+      return 1;
+    }
+    if (a[index0][index1] < b[index0][index1]) {
+      return -1;
+    }
+    return 1;
+  });
+}
+
+const setResetButtonView = (filters, sorter) => {
+  if(sorter.keyName != ''
+      || filters.partOfName != ''
+      || filters.gender != ''
+      || filters.ageRange[0] != minPersonAge
+      || filters.ageRange[1] != maxPersonAge
+      || filters.country != '') {
+    resetButton.classList.remove('off');
+    return;
+  };
+  resetButton.classList.toggle('off',true);
+}
+
+const resetButtonClickHandler = ({filters, sorter}) => {
+
+  if (resetButton.classList.contains('off')) {
+    return;
+  }
+
+  const change = new Event('change');
```

Comment:
Maybe it's better just call the `updateFriendsList` function with the default state instead of dispatching the `change` event by hand